_posts/2025-01-31-wildfly-domain-usage.adoc

```diff
@@ -0,0 +1,523 @@
+{
+"outcome" => "success",
+"result" => undefined,
+"server-groups" => undefined,
+"response-headers" => {"process-state" => "reload-required"}
+}
+
+[domain@embedded /]
+[domain@embedded /] /host=secondary:reload(admin-only=true)
+{"outcome" => "success"}
+----
+
+Please note that if you use tab complete, it should autocomplete your default host name. Giving the host a name can help to analyze the server log output later. Then I need to add another configuration to the `domain-controller`:
+
+[source,xml]
+----
+<domain-controller>
+    <remote authentication-context="myCtx">
+        <discovery-options>
+            <static-discovery name="primary" protocol="${jboss.domain.primary.protocol:remote+http}"
+                              host="${jboss.domain.primary.address}" port="${jboss.domain.primary.port:9990}"/>
+        </discovery-options>
+    </remote>
+</domain-controller>
+----
+
+Here is the equivalent CLI command to do the configuration:
+
+[source,bash]
+----
+/host=secondary:write-attribute(name=domain-controller.remote.authentication-context, value=myCtx)
+----
+
+As the configuration is shown above, I added a `remote` configuration into the `domain-controller` section, and defined the `authentication-context` to be used as `myCtx`. This configuration is different from the legacy authentication. So I need to define this `authentication-context` in the `elytron` subsystem. Here are the details:
+
+[source,xml]
+----
+<profile>
+    <subsystem xmlns="urn:jboss:domain:core-management:1.0"/>
+    <subsystem xmlns="urn:wildfly:elytron:community:18.0" final-providers="combined-providers"
+               disallowed-providers="OracleUcrypto" register-jaspi-factory="false">
+        <authentication-client>
+            <authentication-configuration sasl-mechanism-selector="DIGEST-MD5" name="myConfig"
+                                          authentication-name="admin" realm="ManagementRealm">
+                <credential-reference clear-text="123"/>
+            </authentication-configuration>
+            <authentication-context name="myCtx">
+                <match-rule match-host="${jboss.domain.primary.address}" authentication-configuration="myConfig"/>
+            </authentication-context>
+        </authentication-client>
+        ...
+    </subsystem>
+    ...
+</profile>
+----
+
+Here is the equivalent CLI command to do the configuration:
+
+[source,bash]
+----
+[domain@embedded /] /host=secondary/subsystem=elytron/authentication-configuration=myConfig:add(sasl-mechanism-selector=DIGEST-MD5, authentication-name=admin, realm=ManagementRealm, credential-reference={clear-text="123"})
+{"outcome" => "success"}
+----
+
+[source,bash]
+----
+[domain@embedded /] /host=secondary/subsystem=elytron/authentication-context=myCtx:add(match-rules=[{match-host="${jboss.domain.primary.address}", authentication-configuration=myConfig}])
+{"outcome" => "success"}
+----
```

Comment:
These should be moved above the previous commands. We should create the `myCtx` resource before we assign it the host.

Reply:
Okay! Moved.